File: service_station/service_station/report/service_revenue/service_revenue.js
// Copyright (c) 2023, ameen and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Service Revenue"] = {
	"filters": [
		{
			"fieldname":"ticket_number",
			"label":"Ticket Number",
			"fieldtype":"Link",
			"options":"Parking Ticket",
			"width": 200,
			"reqd": 0
		},
		{
			"fieldname":"inspecting_engineer",
			"label":"Inspecting Engineer",
			"fieldtype":"Link",
			"options":"User",
			"width": 200,
			"reqd": 0
		},
		{
			"fieldname":"plate_number",
			"label":"Plate Number",
			"fieldtype":"Link",
			"options":"Vehicle",
			"width": 200,
			"reqd": 0
		},
		{
			"fieldname":"sales_invoice",
			"label":"Sales Invoice",
			"fieldtype":"Link",
			"options":"Sales Invoice",
			"width": 200,
			"reqd": 0
		},
		{
			"fieldname":"from",
			"label":__("From Date"),
			"fieldtype":"Date",
			"width": 80,
			"reqd": 0,
			"default": dateutil.nowdate()
		},
	]
};

File: service_station/service_station/report/service_revenue/service_revenue.py
# Copyright (c) 2023, ameen and contributors
# For license information, please see license.txt

import frappe


def execute(filters=None):
	columns = get_columns()
	data = get_data(filters)
	return columns, data

def get_data(filters):

	conditions = "1=1"

	if(filters.get("ticket_number")):conditions += f" AND vi.ticket_number='{filters.get('ticket_number')}'"
	if(filters.get("inspecting_engineer")):conditions += f" AND vi.inspecting_engineer='{filters.get('inspecting_engineer')}'"
	if(filters.get("plate_number")):conditions += f" AND vi.vehicle_reg='{filters.get('plate_number')}'"
	print('conditions:',conditions)
	my_query = f"""
		SELECT
			vi.ticket_number,
			vi.inspecting_engineer,
			vi.vehicle_reg,
			si.name,
			si.grand_total
		FROM 
			`tabVehicle Inspection` AS vi
		JOIN
			`tabSales Invoice` AS si
		ON
			si.parking_ticket = vi.ticket_number
		WHERE
			{conditions}
	"""

	data = frappe.db.sql(my_query)
	return data

def get_columns():
	return [
		"Parking Ticket:Link/Parking Ticket:200",
		"Inspecting Engineer:Link/User:200",
		"Plate Number:Link/Vehicle:200",
		"Sales Invoice:Link/Sales Invoice:200",
		"Invoice Amount:Currency:200",
	]